### DISCUSSION TITLE
How to write Hdr file?

### DISCUSSION THREAD

**Discussion by zhouhang95** [image-rs/image#2195] [Q&A] [answered]
```
    let img = image::open("res/newport_loft.jpg").unwrap();
    img.save("newport_loft2.hdr").unwrap();
```
Then, I get this error:
thread 'main' panicked at src\main.rs:335:35:
called `Result::unwrap()` on an `Err` value: Unsupported(UnsupportedError { format: Unknown, kind: Format(Name("Hdr")) })

**Comment by fintelia** [comment]
At the moment you need to directly use [HdrEncoder](https://docs.rs/image/latest/image/codecs/hdr/struct.HdrEncoder.html). Related: #2013

**Reply by zhouhang95** [reply]
Sorry, I'm not good at github workflow. I don't konw how to change this pr.

**Reply by zhouhang95** [reply]
```
    let enc = HdrEncoder::new(Vec::new());
    let data: Vec<Rgb<f32>> = panorama.pixels().cloned().collect();
    enc.encode(&data, panorama.width() as _, panorama.height() as _).unwrap();
```
Now, I can not read encoded data.

**Reply by fintelia** [reply, marked as answer]
You probably want to keep a copy of the Vec you're writing into:
```rust
let mut data = Vec::new();
let enc = HdrEncoder::new(&mut data);
...
